docs/modules/extensions/pages/register.adoc

```diff
@@ -1,26 +1,135 @@
-= Register Extensions
+= Register Processors
 
-== Register one or more extensions
+There are several terms that it's easy to mix up.
 
-These extensions are registered per document using a callback that feels like a DSL:
+User supplied code:
+
+Extension::
+  This is an informal usage.
+Some code that extends or modifies Asciidoctor capabilities by means of Processors.
+Processor::
+  Some code that, when added to Asciidoctor via a Registry, directly extends or alters Asciidoctor behavior by manipulating the source, block tree, or output.
+A processor must in some way extend Asciidoctor::Extensions::Processor.
+Group::
+  Some standalone code that adds Processors to a Registry.
+Groups can be registered globally or associated with a particular Registry.
+Some groups are subclasses of the `Asciidoctor::Extensions::Group` class.
+
+Asciidoctor code:
+
+[#id_registry]
+Registry::
+  A class whose instances track Processors and groups.
+Extensions::
+  The module and singleton object that encapsulates the extension system and provides global capabilities.
+The Object capabilities are provided by the Ruby `class << self` idiom.
+In particular, the Extensions object tracks globally registered groups.
+Extension::
+  A class whose instances delegate to user-provided code.
+ProcessorExtension::
+  An Extension subclass for delegating to Processors.
+
+== Adding processors to a Registry
+
+Processors are added to a Registry by calling a registry method named after the type of processor.
+For instance, to add a block macro, call `registry.blockMacro ...`.
```

Comment:
Well, the document processor method names all end in processor and the syntax ones don't, but I'm going to try mentioning the registration methods on the processor specific pages. More detail may be needed...

Reply:
LGTM

Reply:
@djencks is exactly right. The DSL methods for the syntax processors do not include the term processor because the focus is on the syntax (block, block macro, inline macro). The lifecycle processors need this, otherwise it just doesn't read we'll (pre, tree, post). So I kept the processor term there one so it reads better, but also to distinguish them from syntax processors.